### PR TITLE
feat: Increase unit test coverage across core modules

### LINKED ISSUE #97: feat: Increase unit test coverage across core modules
### Description

Review current unit test coverage across core modules and expand it where needed. Focus on strengthening coverage for key logic and areas with limited or missing tests.

### Motivation

Improve confidence in core behaviour and reduce the risk of regressions as the editor evolves.

### Proposed Implementation

_No response_

### Definition of Done

- [ ] Implementation: Fully implemented according to the Serverless Workflow spec.
- [x] Unit Tests: Comprehensive unit tests are included and passing.
- [ ] Integration Tests: Verified within the monorepo and target environments (Web/VS Code).
- [ ] Documentation: Updated README.md, ADRs, or official docs.
- [ ] Performance: No significant regression in editor responsiveness.
- [ ] Accessibility: UI changes comply with accessibility standards.

### PR DESCRIPTION
**Pull request by kumaradityaraj**
Closes https://github.com/open-workflow-specification/editor/issues/97

## Description

Expand unit test coverage across the diagram editor and supporting utilities by adding tests for component rendering, state transitions, edge cases, error handling, and utility functions. These additions improve confidence in the editor's behavior while covering previously untested branches.

### Sidebar
- Added tests for:
  - Controlled open state changes
  - Mixed rendering scenarios
  - Additional sidebar interactions and edge cases

### Graph utilities
- Added coverage for:
  - Connection rewriting edge cases
  - Parent hierarchy traversal
  - Entry/exit node connection handling
  - Graphs that require no modifications

### Task details
- Added tests for:
  - Mixed field types
  - Nested object flattening edge cases
  - Array handling
  - Optional and missing properties
  - Metadata filtering

### Validation errors
- Added tests covering:
  - Additional validation filtering scenarios
  - Longest matching node ownership
  - General vs node-specific errors
  - Edge cases around field extraction

### Workflow SDK
- Added tests for:
  - Validation error parsing edge cases
  - Duplicate validation error removal
  - Invalid workflow parsing scenarios
  - Prototype pollution sanitization
  - Error handling fallbacks

### DiagramEditorErrorBoundary
- Added coverage for:
  - Error boundary reset behavior
  - Default and custom fallback rendering
  - Non-Error thrown values
  - Reset key edge cases

### DiagramEditor
- Added tests for:
  - Rendering under different editor states
  - Color mode handling
  - Parsing error fallback
  - Context and rendering edge cases

### ErrorSection
- Added tests for:
  - Mixed error items
  - Conditional field rendering
  - Error count rendering
  - Error list rendering

### SidePanel
- Added coverage for:
  - Node details rendering
  - Workflow information rendering
  - Sidebar header fallbacks
  - Selection hint
  - Missing selected node handling
  - Export action visibility

### WorkflowInfoView
- Added tests for:
  - Metadata rendering edge cases
  - Tags-only metadata
  - Empty tags handling
  - General error rendering
  - Optional metadata fields

## Testing

- Added comprehensive unit tests covering rendering, state transitions, error handling, and edge cases.
- Verified all new and existing tests pass successfully.

### CODE REVIEW

File: packages/open-workflow-diagram-editor/tests/diagram-editor/DiagramEditor.test.tsx [hunk 1]
```diff
@@ -71,4 +72,55 @@ describe("DiagramEditor Component", () => {
...
+  });
+
+  it("renders the sidebar provider with the diagram", () => {
+    render(<DiagramEditor content={BASIC_VALID_WORKFLOW_YAML} locale="en" isReadOnly={true} />);
+
+    expect(screen.getByTestId("diagram-container")).toBeInTheDocument();
+    expect(screen.getByRole("button", { name: /toggle sidebar/i })).toBeInTheDocument();
+  });
+
+  it("resets the error boundary when the content changes", () => {
```
Comment (lornakelly) [inline packages/open-workflow-diagram-editor/tests/diagram-editor/DiagramEditor.test.tsx]: This looks the same as the "updates the rendered content when the workflow content changes" test?

**Comment by lornakelly** [comment]
@kumaradityaraj thanks Aditya, just one small comment

**Review by lornakelly — APPROVED** [review]
LGTM

File: packages/open-workflow-diagram-editor/tests/diagram-editor/error-pages/DiagramEditorErrorBoundary.test.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 import { render, screen } from "@testing-library/react";
 import { DiagramEditorErrorBoundary } from "../../../src/diagram-editor/error-pages/DiagramEditorErrorBoundary";
-import { describe, expect, it } from "vitest";
+import { describe, expect, it, vi } from "vitest";
```
Comment (fantonangeli) [inline packages/open-workflow-diagram-editor/tests/diagram-editor/error-pages/DiagramEditorErrorBoundary.test.tsx]: In other testfiles we imported afterEach. I would follow the same style even if vitest is set to global.
```suggestion
import { describe, expect, it, vi, afterEach } from "vitest";
```

**Review by fantonangeli — APPROVED** [review]
LGTM, I only left a small comment, but this PR works well

**Comment by kumaradityaraj** [comment]
@ricardozanini Please review and merge this PR.
